ambry-router/src/main/java/com.github.ambry.router/SimpleOperationTracker.java

```diff
@@ -227,22 +240,32 @@
         replicaPool.add(downReplicas.pollFirst());
       }
     }
+    cloudReplicasPresent =
+        replicaPool.stream().map(ReplicaId::getReplicaType).anyMatch(Predicate.isEqual(ReplicaType.CLOUD_BACKED));
```

Comment:
If `routerGetEligibleReplicasByStateEnabled` is true, the replicaPool won't include cloudReplica. Do we need to explicitly add it into pool?

Reply:
I was under the impression that it would be returned since `CloudServiceClusterChangeHandler` has the following implementation (return the replica unconditionally if it exists):
```
  @Override
  public Stream<AmbryReplica> getReplicaIdsByState(AmbryPartition partition, ReplicaState state) {
    AmbryReplica replica = partitionToReplica.get(partition.toPathString());
    return replica == null ? Stream.empty() : Stream.of(replica);
  }
```

Reply:
Right, I forgot cloud dc is added to `dcToDcInfo`